File: EIDSS v5/android/EIDSS/StartActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using eidssdroid.model.Database;

namespace EIDSS
{
    [Activity(Label = "EIDSS", MainLauncher = true, Icon = "@drawable/icon")]
    public class StartActivity : Activity
    {
        private const int REFLOAD_DIALOG_ID = 0;
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Create your application here
            string pwd;
            using(var db = new EidssDatabase(this))
            {
                pwd = db.Options().strLocalPassword;
            }

            if (string.IsNullOrEmpty(pwd))
            {
                var i = new Intent(ApplicationContext, typeof(SetLocalPasswordActivity));
                StartActivityForResult(i, 2);
            }
            else
            {
                var i = new Intent(ApplicationContext, typeof(LocalLoginActivity));
                StartActivityForResult(i, 1);
            }
        }

        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            if (requestCode == 1)
            {
                if (resultCode == Result.Ok)
                {
                    var i = new Intent(ApplicationContext, typeof(MainActivity));
                    StartActivityForResult(i, 3);
                }
                else
                {
                    Finish();
                }
            }
            if (requestCode == 2)
            {
                if (resultCode == Result.Ok)
                {
                    ShowDialog(REFLOAD_DIALOG_ID);
                }
                else
                {
                    Finish();
                }
            }
            if (requestCode == 3)
            {
                Finish();
            }
            if (requestCode == 4)
            {
                var i = new Intent(ApplicationContext, typeof(MainActivity));
                StartActivityForResult(i, 3);
            }
            base.OnActivityResult(requestCode, resultCode, data);
        }

        protected override Dialog OnCreateDialog(int id)
        {
            switch (id)
            {
                case REFLOAD_DIALOG_ID:
                    return EidssAndroidHelpers.QuestionDialog(this,
                        Resources.GetString(Resource.String.LoadRefOnStart),
                        (sender, args) =>
                        {
                            var i = new Intent(ApplicationContext, typeof(LoadReferencesActivity));
                            StartActivityForResult(i, 4);
                        },
                        (sender, args) =>
                        {
                            var i = new Intent(ApplicationContext, typeof(MainActivity));
                            StartActivityForResult(i, 3);
                        });
            }
            return null;
        }

    }
}

File: EIDSS v5/android.java/workspace/EIDSS/src/com/bv/eidss/StartActivity.java
package com.bv.eidss;

import com.bv.eidss.data.EidssDatabase;
import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;

public class StartActivity extends Activity {
    private final int REFLOAD_DIALOG_ID = 0;

	public StartActivity() {
	}

	@Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        String pwd;
        EidssDatabase db = new EidssDatabase(this);
        pwd = db.Options().getLocalPassword();
        db.close();

        if (pwd == null || pwd.compareTo("") == 0)
        {
			Intent intent = new Intent();
			intent = intent.setClass(getApplicationContext(), SetLocalPasswordActivity.class);
			startActivityForResult(intent, 2);
        }
        else
        {
			Intent intent = new Intent();
			intent = intent.setClass(getApplicationContext(), LocalLoginActivity.class);
			startActivityForResult(intent, 1);
        }
    }
	
	@SuppressWarnings("deprecation")
	@Override
    public void onActivityResult(int requestCode, int resultCode, Intent data)
    {
        if (requestCode == 1)
        {
            if (resultCode == RESULT_OK)
            {
    			Intent intent = new Intent();
    			intent = intent.setClass(getApplicationContext(), MainActivity.class);
    			startActivityForResult(intent, 3);
            }
            else
            {
                finish();
            }
        }
        if (requestCode == 2)
        {
            if (resultCode == RESULT_OK)
            {
                showDialog(REFLOAD_DIALOG_ID);
            }
            else
            {
                finish();
            }
        }
        if (requestCode == 3)
        {
            finish();
        }
        if (requestCode == 4)
        {
			Intent intent = new Intent();
			intent = intent.setClass(getApplicationContext(), MainActivity.class);
			startActivityForResult(intent, 3);
        }
        super.onActivityResult(requestCode, resultCode, data);
    }

	@Override
    protected Dialog onCreateDialog(int id) {
        switch (id)
        {
	        case REFLOAD_DIALOG_ID:
                return EidssAndroidHelpers.QuestionDialog(this,
            		getResources().getString(R.string.LoadRefOnStart),
                	new DialogInterface.OnClickListener(){
						@Override
						public void onClick(DialogInterface dialog, int which) {
							Intent intent = new Intent();
							intent = intent.setClass(getApplicationContext(), LoadReferencesActivity.class);
							startActivityForResult(intent, 4);
						}},
                	new DialogInterface.OnClickListener(){
						@Override
						public void onClick(DialogInterface dialog, int which) {
							Intent intent = new Intent();
							intent = intent.setClass(getApplicationContext(), MainActivity.class);
							startActivityForResult(intent, 3);
						}
					});
        }
        return null;
    }
	
}
